buffer/src/main/java/io/netty/buffer/SlicedByteBuf.java

```diff
@@ -53,8 +61,10 @@ public SlicedByteBuf(ByteBuf buffer, int index, int length) {
             adjustment = index;
         }
         this.length = length;
-
-        writerIndex(length);
+        maxCapacity(length);
+        setIndex(0, length);
+        markReaderIndex();
```

Comment:
Should we wait for https://github.com/netty/netty/pull/3789 and update this to `discardMarks`?